src/renderer.ts

```diff
@@ -69,43 +62,6 @@ export class Renderer {
     return Renderer.buildContext(elementId, Renderer.Backends.SVG, width, height, background);
   }
 
-  // eslint-disable-next-line
-  static bolsterCanvasContext(ctx: any): RenderContext {
-    if (Renderer.USE_CANVAS_PROXY) {
-      return new CanvasContext(ctx);
-    }
-
-    // Modify the CanvasRenderingContext2D to include the following methods, if they do not already exist.
-    // TODO: Is a Proxy object appropriate here?
-    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy
-    const methodNames = [
-      'clear',
-      'setFont',
-      'setRawFont',
-      'setFillStyle',
-      'setBackgroundFillStyle',
-      'setStrokeStyle',
-      'setShadowColor',
-      'setShadowBlur',
-      'setLineWidth',
-      'setLineCap',
-      'openGroup',
-      'closeGroup',
-      'getGroup',
-    ];
-
-    ctx.vexFlowCanvasContext = ctx;
-
-    methodNames.forEach((methodName) => {
-      if (!(methodName in ctx)) {
-        // eslint-disable-next-line
-        ctx[methodName] = (CanvasContext.prototype as any)[methodName];
-      }
-    });
-
-    return ctx;
-  }
```

Comment:
So we are effectively setting USE_CANVAS_PROXY to true from now on? If the performance impact is minimal, then I'm totally on board with the simpler approach. I don't know the history behind USE_CANVAS_PROXY and it is not currently referenced anywhere in the tests.

And in fact, we are already proxying almost all of the method calls, since we modify the CanvasRenderingContext2D to have a new instance var `vexFlowCanvasContext` and then add on new methods which close but not exactly the same as the CanvasRenderingContext2D API's :-D. I guess the main difference is that the VexFlow methods are chainable.

Reply:
I was unable to measure any performance impact. Modern JS engines already inline on hot functions, so if there was any overhead at all, it would come from the code optimizer having to work a tiny bit harder.

The primary reason I tried deleting it to check what the performance impact might be is that `CanvasRenderingContext2D` already has a `measureText` method. So adding a `measureText` method to the `CanvasContext` that called into the `CanvasRenderingContext2D` would require stuffing in a `realMeasureText` attribute too, which just smelt bad.

Following on from the discussion in #1127, I'd like to add a `fillGlyph` method to the context ([I have it written for the `SVGContext` already](https://github.com/tommadams/vexflow/blob/1ee462ee01f1f47314d9ad80e7ebce0cad81c4fa/src/svgcontext.ts#L656)). Once this is added, the canvas context's `fillGlyph` would call the `moveTo`, `lineTo`, `bezierCurveTo`, etc directly on the `CanvasRenderingContext2D` context, which would elide a lot of the wrapped calls.

Reply:
CanvasContext `fillGlyph()` sounds like a good idea! The current loop where we process path instructions and then call moveTo, lineTo, etc isn't ideal. Even if that loop were just moved into CanvasContext, it would remove the proxy overhead, as you noted. But if the SVG paths could somehow be shoved straight into a Canvas Path2D, that'd be extra cool. For example, could we use a single Path2D instance for all of the black noteheads? We might have to make use of a lot of: 
```
    ctx.translate(x,y);
    ctx.stroke(noteheadPath);
    ctx.translate(x,y);
    ctx.stroke(noteheadPath);
```
And probably also `ctx.save()` and `ctx.restore()`.

Reply:
That's a neat idea. You'd need to handle scaling and make sure the context's transform matrix was reset at the end of glyph rendering but it could definitely work.

Reply:
I added `USE_CANVAS_PROXY` when building the VexTab Google Docs plugin -- at the time they had a special canvas implementation which I had to hack around with this to make work. Now that they use a fully-featured runtime, we don't need this anymore (I think.)